kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/service/impl/ProxyAuthServiceImpl.java

```diff
@@ -0,0 +1,648 @@
+		return proxyTokenForLocalEnv(clientSecret.getClientId(), SUCCESS,
+				CLIENTID_AND_TOKEN_COMBINATION_HAD_BEEN_VALIDATED_SUCCESSFULLY);
+	}
+
+	private AuthNResponseDto proxyTokenForLocalEnv(String subject, String status, String message) {
+		long exp = System.currentTimeMillis() + localExp;
+		String token = proxyTokenGenarator.getProxyToken(subject, exp);
+		AuthNResponseDto authNResponseDto = new AuthNResponseDto();
+		authNResponseDto.setToken(token);
+		authNResponseDto.setRefreshToken(null);
+		authNResponseDto.setExpiryTime(exp);
+		authNResponseDto.setStatus(status);
+		authNResponseDto.setMessage(message);
+		return authNResponseDto;
+	}
+
+	/**
+	 * Method used for generating refresh token
+	 * 
+	 * @param existingToken existing token
+	 * 
+	 * @return mosipUserDtoToken is of type {@link MosipUserTokenDto}
+	 * 
+	 * @throws Exception exception
+	 * 
+	 */
+
+	@Override
+	public MosipUserTokenDto retryToken(String existingToken) throws Exception {
+		MosipUserTokenDto mosipUserDtoToken = null;
+		boolean checkRefreshToken = false;
+		AuthToken accessToken = customTokenServices.getTokenDetails(existingToken);
+		if (accessToken != null) {
+			if (accessToken.getRefreshToken() != null) {
+				checkRefreshToken = tokenValidator.validateExpiry(accessToken.getRefreshToken());
+			}
+			if (checkRefreshToken) {
+				TimeToken newAccessToken = tokenGenerator.generateNewToken(accessToken.getRefreshToken());
+				AuthToken updatedAccessToken = customTokenServices.getUpdatedAccessToken(accessToken.getUserId(),
+						newAccessToken, accessToken.getUserId());
+				mosipUserDtoToken = tokenValidator.validateToken(updatedAccessToken.getAccessToken());
+			} else {
+				throw new RuntimeException("Refresh Token Expired");
+			}
+		} else {
+			throw new RuntimeException("Token doesn't exist");
+		}
+		return mosipUserDtoToken;
+	}
+
+	/**
+	 * Method used for invalidate token
+	 * 
+	 * @param token token
+	 * 
+	 * @return authNResponse is of type {@link AuthNResponse}
+	 * 
+	 * @throws Exception exception
+	 * 
+	 */
+
+	@Override
+	public AuthNResponse invalidateToken(String token) throws Exception {
+		AuthNResponse authNResponse = null;
+		customTokenServices.revokeToken(token);
+		authNResponse = new AuthNResponse();
+		authNResponse.setStatus(AuthConstant.SUCCESS_STATUS);
+		authNResponse.setMessage(AuthConstant.TOKEN_INVALID_MESSAGE);
+		return authNResponse;
+	}
+
+	@Override
+	public RolesListDto getAllRoles(String appId) {
+		RolesListDto rolesListDto = keycloakImpl.getAllRoles();
+		return rolesListDto;
+	}
+
+	@Override
+	public MosipUserListDto getListOfUsersDetails(List<String> userDetails, String appId) throws Exception {
+		MosipUserListDto mosipUserListDto = keycloakImpl.getListOfUsersDetails(userDetails);
+		return mosipUserListDto;
+	}
+
+	@Override
+	public MosipUserSaltListDto getAllUserDetailsWithSalt(List<String> userDetails, String appId) throws Exception {
```

Comment:
Should this be based on roles? or we should open this method for everyone?

Reply:
This operation is Unsupported in local profile as John told. The users return by this has relation with masterdata coding all these functionalities will give us to code our IAM again which we removed